pkg/controller/core/clusterqueue_controller.go

```diff
@@ -291,18 +299,43 @@ func (r *ClusterQueueReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *ClusterQueueReconciler) Status(cq *kueue.ClusterQueue) (kueue.ClusterQueueStatus, error) {
+func (r *ClusterQueueReconciler) UpdateCqStatusIfChanged(
+	ctx context.Context,
+	cq *kueue.ClusterQueue,
+	conditionStatus metav1.ConditionStatus,
+	reason, msg string,
+) error {
+	pendingWorkloads := r.qManager.Pending(cq)
 	usage, workloads, err := r.cache.Usage(cq)
 	if err != nil {
 		r.log.Error(err, "Failed getting usage from cache")
 		// This is likely because the cluster queue was recently removed,
 		// but we didn't process that event yet.
-		return kueue.ClusterQueueStatus{}, err
+		return err
 	}
+	if UpdateCqStatus(&cq.Status, pendingWorkloads, workloads, usage, conditionStatus, reason, msg) {
```

Comment:
why split this in two functions?

Reply:
I split this to simplify unit tests for the `UpdateCqStatus` function.

Although, there is no longer any reason to split that function since I would change the test as I mentioned at https://github.com/kubernetes-sigs/kueue/pull/415#discussion_r1047798382.

```
@[alculquicondor](https://github.com/alculquicondor) alculquicondor [yesterday](https://github.com/kubernetes-sigs/kueue/pull/415#discussion_r1046079081)
I'm generally prefer not to test such small functions that are not exported. You could try to test the entire Reconcile or maybe even UpdateCqStatusIfChanged

Member
Author
@[tenzen-y](https://github.com/tenzen-y) tenzen-y [4 minutes ago](https://github.com/kubernetes-sigs/kueue/pull/415#discussion_r1047798382)
SGTM
I would change this test to one using UpdateCqStatusIfChanged.
```

So I would move the processes in the `UpdateCqStatus` func to the `UpdateCqStatusIfChanged` func.

Reply:
Done.